firefox-ios/firefox-ios-tests/Tests/ClientTests/TabManagement/TabManagerTests.swift

```diff
@@ -186,6 +187,26 @@ class TabManagerTests: XCTestCase {
         XCTAssertEqual(inactiveTabs.count, expectedInactiveTabs)
     }
 
+    func test_addTabsForURLs() {
+        let subject = createSubject()
+
+        subject.addTabsForURLs([URL(string: "https://www.mozilla.org/privacy/firefox")!], zombie: false, shouldSelectTab: false)
```

Comment:
@OrlaM @dataports 
When I set `shouldSelectTab` to `true` for `addTabsForURLs`, it hits `selectTab`, which causes the tests to fail due to the memory leaks check. Since I am mainly testing the private functionality as per this bug, I set this to `false` for now. 

I believe there were some changes around `selectTab` that you both may be more aware of, so I thought I would send you a note. Let me know if this is something you would like me to look into further or if I am missing some context. Also, I was unable to find any tests surrounding that method.

Reply:
I'm not sure about this one tbh, will approve for now but would be good to meet with @OrlaM to discuss and see if this needs a ticket to investigate.

Reply:
Created ticket here: 
https://mozilla-hub.atlassian.net/browse/FXIOS-9528

@OrlaM @dataports